### PR TITLE
dbtesting: detect pkg name

### PR DESCRIPTION
**Pull request by keegancsmith**
Using dbtesting is not how new code is written, so direct uses of this should fade away over time. However, it still frustrates me to see its tentacles all over our codebase via the init calls that set DBNameSuffix global.

This is some fun code which eliminates the need to set DBNameSuffix by detecting the package under test and generating the DBNameSuffix based on that.

The first commit contains the meat of the change. The second commit is larger and is a mechanical removal of DBNameSuffix since it is no longer read.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Neat idea!

File: internal/database/dbtesting/dbtesting.go [hunk 1]
```diff
@@ -177,6 +171,67 @@ func initTest(nameSuffix string) error {
...
+	if len(pkg) > 32 {
+		pkg = fmt.Sprintf("%X-%s", crc32.ChecksumIEEE([]byte(pkg)), pkg[len(pkg)-32:])
+	}
+
+	return "sourcegraph-test-" + strings.ReplaceAll(pkg, "/", "-"), nil
+}
+
+// testPkgName finds the relative name of the sourcegraph package being tested
+// by inspecting the call stack. If it fails, it returns an empty string.
+func testPkgName() string {
```
Comment (camdencheek) [inline internal/database/dbtesting/dbtesting.go:192]: I like this much better than [hashing the working directory](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/dbtest/dbtest.go?L170:6). Maybe we can use this in `dbtest` as well to get better test database names.

**Review by camdencheek — APPROVED** [review]
Super glad to see all those inits going away 🥇

**Comment by keegancsmith** [comment]
> Maybe we can use this in `dbtest` as well to get better test 
> database names.

SGTM

**Pull request merged**